app/src/main/kotlin/at/bitfire/davdroid/repository/AccountRepository.kt

```diff
@@ -216,14 +215,18 @@ class AccountRepository @Inject constructor(
             serviceRepository.renameAccount(oldName, newName)
 
             // update address books
-            localAddressBookStore.get().updateAccount(oldAccount, newAccount)
+            val contactsProvider = context.contentResolver.acquireContentProviderClient(ContactsContract.AUTHORITY)
+            localAddressBookStore.get().updateAccount(contactsProvider!!, oldAccount, newAccount)
 
             // calendar provider doesn't allow changing account_name of Events
             // (all events will have to be downloaded again at next sync)
 
             // update account_name of local tasks
             try {
-                LocalTaskList.onRenameAccount(context, oldAccount.name, newName)
+                TaskProvider.acquire(context)?.use { provider ->
```

Comment:
`TaskProvider.acquire` is not a DAVx5 function, but from ical4android. So it doesn't know anything about the tasks app selection in DAVx5. If no `ProviderName` is given to it – like here – it tries all providers, and if there are multiple providers installed, it can very well be that this is not the same provider as `taskAppManager.currentProvider()`. Then the `dataStore` tasks provider won't match the acquired content provider and the operation would fail.

I think the problem arises because we shouldn't have to know anything about the (current) tasks app here. Everything should be done be `TasksAppManager`, hence the `getDataStore()` call.

Unfortunately, the `DataStore` takes a `ContentProviderClient` for it's operations. Possible solutions:

- We could acquire the content provider client within the data store and keep it there. In this case the `DataStore` would have to be `Closeable` and close the provider client.
- As a solution inbetween, we could make `DataStore.acquireClient()`, which returns the correct content provider client. We can then `use()` it in `Syncer` and `updateAccount()`. Then we don't need to explicitly acquire anything here.